### PR TITLE
Some minor fixes to the SuiteHeader component

### PR DESCRIPTION
**Pull request by blechner**
- Fixing a value in `ROUTE_TYPES` (copy/paste error)
- Adding a `useEffect` to capture changes in `surveyData` after the component is mounted.
- Adding a `z-index` to the toast notification to make sure the links are always clickable.

### CODE REVIEW

**Review by tay1orjones — APPROVED** [review]

File: src/components/SuiteHeader/_suite-header.scss [hunk 1]
```diff
@@ -59,6 +59,7 @@
   right: 0;
   max-width: rem(425px);
   width: auto;
+  z-index: 1;
```
Comment (tay1orjones) [inline src/components/SuiteHeader/_suite-header.scss:62]: There's a z-index function that we can use to keep values consistent, it takes variables as params - [here's the source/list](
https://github.com/carbon-design-system/carbon/blob/73d94961ba828afc72fbb82a853d304caffbd023/packages/components/src/globals/scss/_layout.scss#L14-L23)


```suggestion
  z-index: z('overlay');
```

**Review by tay1orjones — APPROVED** [review]
One minor comment but overall LGTM

**Review by cgirani — APPROVED** [review]

File: src/components/SuiteHeader/_suite-header.scss (hunk 1 above)
Comment (blechner) [reply src/components/SuiteHeader/_suite-header.scss:62]: Thanks for pointing this out @tay1orjones 
I didn't know about this `z-index` function 👍

**Pull request merged**